File: c/bc2844.c
#include <stdio.h>
#include <string.h>

int typingTime(int lag, int react, int timeSpent, int lenStr) {
    return 2*lag + react + (timeSpent * lenStr);
}

int compare_int(int a, int b) {
    return (a > b) - (a < b);
}

void winner(int ttM, int ttV) {
    int x = compare_int(ttM, ttV);
    if(x == 1) printf("Vinicius\n");
    else if (x == -1) printf("Matheus\n");
    else printf("Empate\n");
}

int main(void) {
    int lagMat, reactMat, timeSpentMat, lagVini, reactVini, timeSpentVini;
    int lenStr, totalTimeMat, totalTimeVini;
    char stringDuel[100001] = {0}, line[100];

    fgets(line, sizeof(line), stdin);
    sscanf(line, "%d %d %d", &lagMat, &reactMat, &timeSpentMat);
    fgets(line, sizeof(line), stdin);
    sscanf(line, "%d %d %d", &lagVini, &reactVini, &timeSpentVini);

    fgets(stringDuel, sizeof(stringDuel), stdin);
    lenStr = strlen(stringDuel);
    while(stringDuel[lenStr-1] == '\n' || stringDuel[lenStr-1] == '\r')
        stringDuel[--lenStr] = '\0';

    totalTimeMat = typingTime(lagMat, reactMat, timeSpentMat, lenStr);
    totalTimeVini = typingTime(lagVini, reactVini, timeSpentVini, lenStr);

    winner(totalTimeMat, totalTimeVini);

    return 0;
}

File: java/src/main/BC2844.java
import java.util.Scanner;
import java.util.Arrays;

public class BC2844 {

    public static int[] readValues(Scanner scanner){
    String line = scanner.nextLine();

    return Arrays.stream(line.split(" "))
                          .mapToInt(Integer::parseInt)
                          .toArray();
    }

    public static int typingTime(int lag, int react, int timeSpent, int lenStr) {
        return 2*lag + react + timeSpent * lenStr;
    }

    public static void winner(int timeMat, int timeVini){
        int x = Integer.compare(timeMat, timeVini);
        if(x == -1) System.out.println("Matheus");
        else if(x == 1) System.out.println("Vinicius");
        else System.out.println("Empate");
    }

    public static void main(String args[]){
        Scanner scanner = new Scanner(System.in);

        int[] matValues = readValues(scanner);
        int[] viniValues = readValues(scanner);

        int lagMat = matValues[0];
        int reactMat = matValues[1];
        int timeSpentMat = matValues[2];

        int lagVini = viniValues[0];
        int reactVini = viniValues[1];
        int timeSpentVini = viniValues[2];

        String sentence = scanner.nextLine();
        int lenSentence = sentence.length();

        int totalTimeMat = typingTime(lagMat, reactMat, timeSpentMat, lenSentence);
        int totalTimeVini = typingTime(lagVini, reactVini, timeSpentVini, lenSentence);

        winner(totalTimeMat, totalTimeVini);

        scanner.close();
    }
}